savant/utils/image.py

```diff
@@ -0,0 +1,528 @@
+                    :,
+                ]
+                res_start_col = max(-start_col, 0)
+                res_start_row = max(-start_row, 0)
+                res_image[
+                    res_start_row : res_start_row + cut_roi.size()[1],
+                    res_start_col : res_start_col + cut_roi.size()[0],
+                    :,
+                ] = cut_roi
+            else:
+                res_image = self._np_image[start_row:end_row, start_col:end_col, :]
+            res_bbox = RBBox(
+                x_center=res_image.shape[1] / 2,
+                y_center=res_image.shape[0] / 2,
+                width=bbox.width,
+                height=bbox.height,
+                angle=bbox.angle,
+            )
+
+            return CPUImage(res_image), res_bbox
+        raise ValueError(f'Unknown bbox type {type(bbox)}')
+
+    def paste(self, image: 'CPUImage', point: Tuple[int, int]):
+        """Pastes image on current image.
+
+        :param image: image to paste
+        """
+        self._np_image[
+            point[1] : point[1] + image.height, point[0] : point[0] + image.width
+        ] = image._np_image
+
+    def concat(self, image: 'CPUImage', axis: int = 0) -> 'CPUImage':
+        """Concatenates images along axis.
+
+        :param image: image to concatenate
+        :param axis: axis to concatenate. 0 - is vertical, 1 - is horizontal
+        :return: concatenated image
+        """
+        assert self._np_image.dtype == image.np_array.dtype, (
+            f'Images have different types {self._np_image.dtype } != '
+            f'{image.np_array.dtype}'
+        )
+        if axis == 0:
+            assert (
+                self.width == image.width
+            ), f'Images have different height {self.width} != {image.height}'
+
+            return CPUImage(
+                image=np.concatenate([self._np_image, image.np_array], axis=0)
+            )
+        elif axis == 1:
+            assert (
+                self.height == image.height
+            ), f'Images have different width {self.height} != {image.height}'
+            return CPUImage(
+                image=np.concatenate([self._np_image, image.np_array], axis=1)
+            )
+        else:
+            raise ValueError(f'Unknown axis {axis}')
+
+    def rotate(
+        self, angle: float, bbox: Optional[RBBox] = None
+    ) -> Tuple['CPUImage', RBBox]:
+        """Rotates image on angle.
+
+        :param angle: angle to rotate in degrees
+        :param bbox: image will be rotated on bbox.angle if bbox is not None
+        :return: rotated image
+        """
+
+        if bbox is not None:
+            rotation_matrix, resolution = get_rotation_matrix(
+                self, angle, (bbox.x_center, bbox.y_center)
+            )
+        else:
+            rotation_matrix, resolution = get_rotation_matrix(
+                self, angle, (self.width // 2, self.height // 2)
+            )
```

Comment:
```suggestion
        if bbox is not None:
            rotation_point = bbox.x_center, bbox.y_center
        else:
            rotation_point = self.width / 2, self.height / 2
        rotation_matrix, resolution = get_rotation_matrix(
                self, angle, rotation_point
        )
```

Reduce duplication, also rotation point is a tuple of floats, so no need for integer division